user_tools/src/spark_rapids_pytools/rapids/rapids_tool.py

```diff
@@ -608,7 +609,7 @@ def populate_dependency_list() -> List[RuntimeDependency]:
             # check if the dependencies is defined in a config file
             config_obj = self.get_tools_config_obj()
             if config_obj is not None:
-                if config_obj.runtime.dependencies:
+                if config_obj.runtime and config_obj.runtime.dependencies:
```

Comment:
Since `runtime` field in the tools config has been made optional, we need to check if `config_obj.runtime` is not None.